pulsar-client-go/pulsar/c_message.go

```diff
@@ -97,8 +98,8 @@ func buildMessage(message ProducerMessage) *C.pulsar_message_t {
 
 ////////////// Message
 
-func newMessageWrapper(ptr *C.pulsar_message_t) Message {
-	msg := &message{ptr: ptr}
+func newMessageWrapper(schema Schema, ptr *C.pulsar_message_t) Message {
```

Comment:
just need some clarifications from you: is `schema` here a reference copy or a value copy?

Reply:
Interfaces are always passed by ref. Go syntax is a bit confusing since you need to know that it's indeed an interface and not a struct